test_scripts/Capabilities/PersistingHMICapabilities/common.lua

```diff
@@ -0,0 +1,419 @@
+  local data = m.getCacheCapabilityTable()
+  if data.VR.language == pLanguage and data.TTS.language == pLanguage and data.UI.language == pLanguage then
+    m.print(35, "Languages were changed")
+  else
+    m.failTestStep("SDL doesn't updated cache file")
+  end
+end
+
+function m.updateHMILanguage(pLanguage)
+  local hmiValues = m.getDefaultHMITable()
+  hmiValues.UI.GetLanguage.params.language = pLanguage
+  hmiValues.VR.GetLanguage.params.language = pLanguage
+  hmiValues.TTS.GetLanguage.params.language = pLanguage
+  return hmiValues
+end
+
+function m.registerApp(pAppId, pCapResponse, pMobConnId, hasPTU)
+  if not pCapResponse then pCapResponse = {} end
+  if not pAppId then pAppId = 1 end
+  if not pMobConnId then pMobConnId = 1 end
+
+  local policyModes = {
+    P  = "PROPRIETARY",
+    EP = "EXTERNAL_PROPRIETARY",
+    H  = "HTTP"
+  }
+
+  local session = m.createSession(pAppId, pMobConnId)
+  session:StartService(7)
+  :Do(function()
+     local msg = ""
+      local corId = session:SendRPC("RegisterAppInterface", m.getParams(pAppId))
+      m.getHMIConnection():ExpectNotification("BasicCommunication.OnAppRegistered",
+        { application = { appName = m.getParams(pAppId).appName } })
+      :Do(function(_, d1)
+          m.setHMIId(d1.params.application.appID, pAppId)
+          if hasPTU then
+            m.ptu.expectStart()
+          end
+        end)
+      session:ExpectResponse(corId, { success = true, resultCode = "SUCCESS" }  )
+      :ValidIf(function(_,data)
+        for key, value in pairs (pCapResponse) do
+          print(key)
+          if not m.isTableEqual(data.payload[key], pCapResponse[key]) then
+            msg = msg .. key.." contains unexpected parameters\n"..
+              " Expected: " .. m.tableToString(pCapResponse[key]) .. "\n" ..
+              " Actual: " .. m.tableToString(data.payload[key]) .. "\n"
+          end
+        end
+        if string.len(msg) > 0 then
+          return false, msg
+        else
+          return true
+        end
+      end)
+      :Do(function()
+          session:ExpectNotification("OnHMIStatus",
+            { hmiLevel = "NONE", audioStreamingState = "NOT_AUDIBLE", systemContext = "MAIN" })
+          session:ExpectNotification("OnPermissionsChange")
+          :Times(AnyNumber())
+          local policyMode = SDL.buildOptions.extendedPolicy
+          if policyMode == policyModes.P or policyMode == policyModes.EP then
+            session:ExpectNotification("OnSystemRequest", { requestType = "LOCK_SCREEN_ICON_URL" })
+          end
+        end)
+    end)
+end
+
+function m.masterReset(pExpFunc)
+  local isOnSDLCloseSent = false
+  if pExpFunc then pExpFunc() end
+  m.getHMIConnection():SendNotification("BasicCommunication.OnExitAllApplications", { reason = "MASTER_RESET" })
+  m.getHMIConnection():ExpectNotification("BasicCommunication.OnSDLClose")
+  :Do(function()
+    isOnSDLCloseSent = true
+    SDL.DeleteFile()
+  end)
+  :Times(AtMost(1))
```

Comment:
@VjKlepikov Please investigate the reason of `Times(AtMost(1))` here.

Reply:
done